static/app/views/insights/common/components/releaseSelector.tsx

```diff
@@ -134,8 +159,48 @@ function LabelDetails(props: LabelDetailsProps) {
   );
 }
 
+function getSortRealesesBy(
+  sort: ReleasesSortByOption,
+  environments: string[]
+): ReleasesSortByOption {
+  // Require 1 environment for date adopted
+  if (sort === ReleasesSortOption.ADOPTION && environments.length !== 1) {
+    return ReleasesSortOption.DATE;
+  }
+
+  const sortExists = Object.keys(SORT_BY_OPTIONS).includes(sort ?? '');
+
+  if (sortExists) {
+    return sort;
+  }
+
+  return ReleasesSortOption.DATE;
+}
```

Comment:
you can find the same logic [here](https://github.com/getsentry/sentry/blob/5078825918dfc42a0fd3cccbaa3643fdbc776d15/static/app/views/releases/list/index.tsx#L148-L163)